src/views/Browse/components/HeaderCell.tsx

```diff
@@ -0,0 +1,44 @@
+import {
+  faArrowsAltV,
+  faLongArrowAltDown,
+  faLongArrowAltUp,
+} from "@fortawesome/free-solid-svg-icons";
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import { useMemo } from "react";
+
+import { SortingColumnName, useSort } from "../providers";
+
+export const HeaderCell: React.FC<{
+  className?: string;
+  name: SortingColumnName;
+  sortable?: boolean;
+}> = ({ children, className, name, sortable = true }) => {
+  const { sortColumn, sorting } = useSort();
+
+  const [icon, iconColor] = useMemo(() => {
+    if (sorting && sorting.column === name) {
+      return sorting.direction === "asc"
+        ? [faLongArrowAltUp, "text-white"]
+        : [faLongArrowAltDown, "text-white"];
+    }
+    return [faArrowsAltV, "text-gray-600"];
+  }, [name, sorting]);
```

Comment:
This is another example of `useMemo/Callback` overuse as this is not only less readable but also slower. For example, will it be more readable when expressed this way?

```ts
const [ icon, iconColor] = pickColors(sorting, name)


// it does not need to be declared inside the render function
function pickColors(sorting, name){
  if (sorting && sorting.column === name) {
    return sorting.direction === "asc"
      ? [faLongArrowAltUp, "text-white"]
      : [faLongArrowAltDown, "text-white"];
  }
  
  return [faArrowsAltV, "text-gray-600"];
}
```

Reply:
After the useCallback/useMemo refactor, I ended up with [pretty much exactly that](https://github.com/tadeaspetak/teleport-challenge/blob/18066fd55ecec2c6be4ec78fe77434ea5c6f7169/src/views/Browse/components/HeaderCell.tsx#L11).